eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/ZmqWatcher.scala

```diff
@@ -226,13 +238,132 @@ class ZmqWatcher(chainHash: ByteVector32, blockCount: AtomicLong, client: Extend
   // CHANGING THIS WILL RESULT IN CONCURRENCY ISSUES WHILE PUBLISHING PARENT AND CHILD TXS
   val singleThreadExecutionContext = ExecutionContext.fromExecutor(Executors.newSingleThreadExecutor())
 
-  def publish(tx: Transaction, isRetry: Boolean = false): Unit = {
-    log.info(s"publishing tx (isRetry=$isRetry): txid=${tx.txid} tx={}", tx)
-    client.publishTransaction(tx)(singleThreadExecutionContext).recover {
+  def publish(p: PublishAsap): Future[ByteVector32] = {
+    p.strategy match {
+      case PublishStrategy.SetFeerate(currentFeerate, targetFeerate, dustLimit, signingKit) =>
+        val spentOutpoint = signingKit match {
+          case signingKit: TransactionSigningKit.ClaimAnchorOutputSigningKit => signingKit.txWithInput.input.outPoint
+          case signingKit: TransactionSigningKit.HtlcTxSigningKit => signingKit.txWithInput.input.outPoint
+        }
```

Comment:
Should the `outPoint` be exposed in `TransactionSigningKit`?

Reply:
I hesitated...it's a bit early to say, in the longer run I'll provide signing kits for every tx (claim-htlc-tx, 3rd-stage txs, main-delayed, etc) so I don't know yet what will make sense in that abstraction. I can add it for now, that would make sense.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1702/commits/5d88187901aa9a1892726540ff68d82dc77a7961